teps/0040-ignore-step-errors.md

```diff
@@ -145,13 +154,236 @@ This proposal is limited to a step within a task and does not address `pipelineT
+      "message": "All Steps have completed executing",
+      "reason": "Succeeded",
+      "status": "True",
+      "type": "Succeeded"
+    }
+  ],
+  "podName": "taskrun-unit-test-t6qcl-pod-zpqs9",
+  "startTime": "2021-06-21T18:21:57Z",
+  "steps": [
+    {
+      "container": "step-ignore-unit-test-failure",
+      "imageID": "...",
+      "name": "ignore-unit-test-failure",
+      "terminated": {
+        "containerID": "...",
+        "exitCode": 1,
+        "finishedAt": "2021-06-21T18:22:05Z",
+        "reason": "Completed",
+        "startedAt": "2021-06-21T18:22:05Z"
+      }
+    },
+  ],
+```
+
+In the following example, the `pipelineRun` is executing two tasks with the first task defined to ignore the step error.
+If that step is able to produce a result file before failing, that result is made available to it's consuming task.
+
+#### `pipelineRun` with a failing step
+
+```json
+kubectl get pr pipelinerun-with-failing-step-mdncp -o json | jq .status.taskRuns | jq 'map(.status)'
+[
+  {
+    "completionTime": "2021-06-21T18:47:40Z",
+    "conditions": [
+      {
+        "lastTransitionTime": "2021-06-21T18:47:40Z",
+        "message": "All Steps have completed executing",
+        "reason": "Succeeded",
+        "status": "True",
+        "type": "Succeeded"
+      }
+    ],
+    "podName": "pipelinerun-with-failing-step-mdncp-task1-7gvl5-pod-dz5hb",
+    "startTime": "2021-06-21T18:47:32Z",
+    "steps": [
+      {
+        "container": "step-write-a-result",
+        "imageID": "...",
+        "name": "write-a-result",
+        "terminated": {
+          "containerID": "...",
+          "exitCode": 11,
+          "finishedAt": "2021-06-21T18:47:39Z",
+          "message": "[{\"key\":\"task1-result\",\"value\":\"123\",\"type\":\"TaskRunResult\"}]",
+          "reason": "Completed",
+          "startedAt": "2021-06-21T18:47:39Z"
+        }
+      }
+    ],
+    "taskResults": [
+      {
+        "name": "task1-result",
+        "value": "123"
+      }
+    ],
+    ...
+  },
+  {
+    "completionTime": "2021-06-21T18:47:49Z",
+    "conditions": [
+      {
+        "lastTransitionTime": "2021-06-21T18:47:49Z",
+        "message": "All Steps have completed executing",
+        "reason": "Succeeded",
+        "status": "True",
+        "type": "Succeeded"
+      }
+    ],
+    "podName": "pipelinerun-with-failing-step-mdncp-task2-9w7cj-pod-sw6sw",
+    "startTime": "2021-06-21T18:47:40Z",
+    "steps": [
+      {
+        "container": "step-verify-a-task-result",
+        "imageID": "...",
+        "name": "verify-a-task-result",
+        "terminated": {
+          "containerID": "...",
+          "exitCode": 0,
+          "finishedAt": "2021-06-21T18:47:49Z",
+          "reason": "Completed",
+          "startedAt": "2021-06-21T18:47:49Z"
+        }
+      }
+    ],
+    ...
+  }
+]
+```
+
+This new field `exit` will be implemented as a `alpha` feature and can be enabled by setting `enable-api-fields`
```

Comment:
nit: i think this needs to be updated to match the new syntax

Reply:
nope, this is updated 😄  It has `exitCode` before and is updated to `exit` now. The new syntax is introducing a section `exit` which in turn has `continue`. The `exit` section will be implemented as `alpha` feature.

Reply:
I though the new feature API field was `onError` ?